R/md5.R

```diff
@@ -0,0 +1,14 @@
+
+md5 <- function(x, raw = TRUE, seed = 0L, ...) {
+  UseMethod("md5")
+}
+
+md5.default <- function(x, raw = TRUE, seed = 0L, ...) {
+  digest(x, algo = "md5", raw = raw, seed = seed, ...)
+}
+
+md5.raw <- function(x, raw = TRUE, seed = 0L, ...) .Call(md5_impl, x, raw, seed)
+
+md5.integer <- function(x, raw = TRUE, seed = 0L, ...) .Call(md5_impl, x, raw, seed)
+
+md5.numeric <- function(x, raw = TRUE, seed = 0L, ...) .Call(md5_impl, x, raw, seed)
```

Comment:
This is 'best practives' for R to dispatch on type ... but given that the right hand side is in fact identical and _still needs the TYPEOF switch in the compiled code_ does not really buy us anything, right?

And we also want to support `character`. And/or `list`. And/or `data.frame`.  So why not just dispatch 'everything' as digest() does?   (Not a trick question, am generally curious...)

Reply:
yeah, i original went into the approach unaware that it could be picked up C side - now it's really more understanding what types can we just do vs which do we have to send to digest.

i do think we get rid of this if there's a generic serializer on the C side.

Reply:
We're on the same page here.